cpp/src/io/csv/csv_gpu.cu

```diff
@@ -1019,21 +1019,30 @@ thrust::host_vector<column_type_histogram> detect_column_types(
   const int block_size = csvparse_block_dim;
   const int grid_size  = (row_starts.size() + block_size - 1) / block_size;
 
-  auto d_stats = rmm::device_vector<column_type_histogram>(num_active_columns);
+  auto d_stats = rmm::device_uvector<column_type_histogram>(num_active_columns, stream);
+  CUDA_TRY(cudaMemsetAsync(
+    d_stats.data(), 0, num_active_columns * sizeof(column_type_histogram), stream.value()));
 
   data_type_detection<<<grid_size, block_size, 0, stream.value()>>>(
     options, data, column_flags, row_starts, d_stats);
 
-  return thrust::host_vector<column_type_histogram>(d_stats);
+  std::vector<column_type_histogram> h_stats(num_active_columns);
```

Comment:
Use make_std_vector_*. (Not merged yet)

Reply:
I can't find the PR that adds this. Will wait till next week to continue with this PR.

Reply:
#7758

Reply:
The utility has been merged in #7798